src/Microsoft.Diagnostics.Monitoring/FileStreamWrapper.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    /// <summary>
+    /// We want to make sure we destroy files we finish streaming.
+    /// We want to make sure that we stream out files since we compress on the fly; the size cannot be known upfront.
+    /// CONSIDER The above implies knowledge of how the file is used by the rest api.
+    /// </summary>
+    internal sealed class FileStreamWrapper : FileStream
```

Comment:
What's the point of thus type if it's only used in a single place?

Reply:
Agreed with David that if it is only single use its probably not worth having an abstraction for it. However if we do keep it I'd suggest we rename it so the purpose is more apparent. Perhaps 'AutoDeleteFileStream'?